### ISSUE TITLE
Progress Bars 

### ISSUE THREAD

**Issue by omer54321** [keras-team/keras#12349]
So, I created a simple CNN for the MNIST dataset, and when I am training it with `verbose = 1`, I suppose I should be getting progress bars that fill up over time (Correct me if I'm wrong here). Instead of  updating, I the progress bar is just getting printed over and over again until the training is done. Example:
`
 1824/60000 [..............................] - ETA: 1:12 - loss: 1.5713 - acc: 0.5620
 1888/60000 [..............................] - ETA: 1:12 - loss: 1.5413 - acc: 0.5683
 1952/60000 [..............................] - ETA: 1:12 - loss: 1.5180 - acc: 0.5743
 2016/60000 [>.............................] - ETA: 1:11 - loss: 1.4878 - acc: 0.5813
 2080/60000 [>.............................] - ETA: 1:11 - loss: 1.4603 - acc: 0.5885
`
 And so on. I am getting this both on Pycharm and through the python console (even moved the .py file outside the folder, so I'd be using Tensorflow and Keras I manually installed and verified they're up to date.
Any clues?
P.S. Here is my code: https://gist.github.com/omer54321/bde99aa48794a4221fd308a2c55a33dc

**Comment by gabrieldemarmiesse** [comment]
This is a pycharm issue, not a keras issue. It works fine in any terminal. So I don't think we should fix it. Please reach out to the pycharm team if you want them to support it.

**Comment by msaroufim** [comment]
We could potentially support progress bars natively in Keras. Something like the below should work

`# Callback allowing to display the progression of the training task.
class ProgressTracker(keras.callbacks.Callback):

  def __init__(self, num_epochs, step=5):
    self.num_epochs = num_epochs
    self.current_epoch = 0.
    self.step = step
    self.last_percentage_report = 0

  def on_epoch_end(self, batch, logs={}):
    self.current_epoch += 1.
    training_percentage = int(self.current_epoch * 100.0 / self.num_epochs)
    if training_percentage - self.last_percentage_report >= self.step:
      print('Training ' + str(
          training_percentage) + '% complete. Training loss: ' + str(
              logs.get('loss')) + ' | Validation loss: ' + str(
                  logs.get('val_loss')))
      self.last_percentage_report = training_percentage`
